### PR TITLE
refactor tsconfigs

### PR DESCRIPTION
**Pull request by dannycoates**
Extends #5430 to refactor all tsconfig files and update `require` paths and moves all custom ts type declarations to `/types`

### CODE REVIEW

**Comment by dannycoates** [comment]
Adding @jodyheavener for review since you seem stoked about it :)

File: packages/fxa-auth-server/lib/geodb.js [hunk 1]
```diff
@@ -5,7 +5,7 @@
 'use strict';
 
 const config = require('../config').get('geodb');
-const geodb = require('../../fxa-geodb')(config);
+const geodb = require('fxa-geodb')(config);
```
Comment (jodyheavener) [inline packages/fxa-auth-server/lib/geodb.js:8]: How is this working? Does node know to resolve this somewhere (like fxa-react), and if so can this change be extended to the other relative require statements?

File: types/fxa-jwtool/index.d.ts [hunk 2]
```diff
@@ -177,7 +177,7 @@ declare namespace fxa_jwtool {
 
   type PublicKeyType = {
     e: string;
-    'fxa-createdAt'?: number;
+    "fxa-createdAt"?: number;
```
Comment (jodyheavener) [inline types/fxa-jwtool/index.d.ts:180]: Is prettier skipping this file?

File: packages/fxa-auth-server/lib/geodb.js (hunk 1 above)
Comment (dannycoates) [reply packages/fxa-auth-server/lib/geodb.js:8]: it's resolve through package.json using `"fxa-geodb": "workspace:*"` and yes we should be using this style everywhere. idk why i didn't catch them all with my grep

File: types/fxa-jwtool/index.d.ts (hunk 2 above)
Comment (dannycoates) [reply types/fxa-jwtool/index.d.ts:180]: weird, idk

**Pull request merged**